rir/src/compiler/translations/rir_2_pir/rir_2_pir.cpp

```diff
@@ -437,11 +449,8 @@ bool Rir2Pir::compileBC(
         if (typeFeedback[rhs].numTypes > 0 &&                                  \
             typeFeedback[lhs].numTypes > 0 &&                                  \
             !typeFeedback[rhs].observedObject() &&                             \
-            !typeFeedback[lhs].observedObject()) {                             \
-            Value* leftIsObj = insert(new IsObject(lhs));                      \
-            insert.conditionalDeopt(leftIsObj, srcCode, pos, stack, false);    \
-            Value* rightIsObj = insert(new IsObject(rhs));                     \
-            insert.conditionalDeopt(rightIsObj, srcCode, pos, stack, false);   \
+            !typeFeedback[lhs].observedObject() && deoptIfObject(lhs) &&       \
+            deoptIfObject(rhs)) {                                              \
```

Comment:
What about the mixed case? When `lhs` gives true and `rhs` false? Won't we have an unnecessary deopt then?

Reply:
I know. but ok, gonna write it differently, you are right